rustls/src/client/hs.rs

```diff
@@ -1181,6 +1189,33 @@ impl State<ClientConnectionData> for ExpectServerHelloOrHelloRetryRequest {
     }
 }
 
+#[derive(Debug)]
+struct CertificateTypeExtensionParams {
```

Comment:
I don't think defining this type adds much value. It would be better to just define a function, maybe called `check_certificate_type()`.